internal/powerplatform/services/data_record/resource_data_record.go

```diff
@@ -0,0 +1,223 @@
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"environment_id": schema.StringAttribute{
+				Description: "Id of the Dynamics 365 environment",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"table_logical_name": schema.StringAttribute{
+				Description: "Name of the data record table",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"columns": schema.DynamicAttribute{
+				Description: "Columns of the data record table",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func (r *DataRecordResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	clientApi := req.ProviderData.(*api.ProviderClient).Api
+	if clientApi == nil {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *http.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+	r.DataRecordClient = NewDataRecordClient(clientApi)
+}
+
+func (r *DataRecordResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var state *DataRecordResourceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &state)...)
+
+	var plan DataRecordResourceModel
+	resp.State.Get(ctx, &plan)
+
+	tflog.Debug(ctx, fmt.Sprintf("CREATE RESOURCE START: %s", r.ProviderTypeName))
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	plan.Id = types.StringValue(plan.Id.ValueString())
+	plan.EnvironmentId = types.StringValue(plan.EnvironmentId.ValueString())
+	plan.TableLogicalName = types.StringValue(plan.TableLogicalName.ValueString())
+	plan.Columns = types.DynamicValue(plan.Columns)
+
+	var mapColumns map[string]interface{}
+
+	jsonColumns, _ := json.Marshal(plan.Columns.String())
+	unquotedJsonColumns, _ := strconv.Unquote(string(jsonColumns))
+	json.Unmarshal([]byte(unquotedJsonColumns), &mapColumns)
+
+	dr, err := r.DataRecordClient.ApplyDataRecord(ctx, plan.Id.ValueString(), plan.EnvironmentId.ValueString(), plan.TableLogicalName.ValueString(), mapColumns)
+	if err != nil {
+		resp.Diagnostics.AddError(fmt.Sprintf("Client error when creating %s", r.ProviderTypeName), err.Error())
+		return
+	}
+
+	plan.Id = types.StringValue(dr.Id)
+
+	tflog.Trace(ctx, fmt.Sprintf("created a resource with ID %s", plan.TableLogicalName.ValueString()))
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+
+	tflog.Debug(ctx, fmt.Sprintf("CREATE RESOURCE END: %s", r.ProviderTypeName))
+}
+
+func (r *DataRecordResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state *DataRecordResourceModel
+
+	tflog.Debug(ctx, fmt.Sprintf("READ RESOURCE START: %s", r.ProviderTypeName))
```

Comment:
why read does not read anything from that backend?

Reply:
done!